src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -165,6 +170,97 @@ Literal Literal::make(std::string_view lexical_form, const IRI &datatype, Node::
+
+    if (this->null()) {
+        return Literal{};
+    }
+
+    auto const this_dtid = this->datatype_id();
+    DatatypeIDView const target_dtid{target};
+
+    if (this_dtid == target_dtid) {
+        return static_cast<Literal>(this->to_node_storage(node_storage));
+    }
+
+    if (target_dtid == Boolean::datatype_id) {
+        // any -> bool
+        return this->ebv_as_literal(node_storage);
+    }
+
+    if (target_dtid == String::datatype_id) {
+        // any -> string
+        return this->as_lexical_form(node_storage);
+    }
+
+    if (this_dtid == String::datatype_id) {
+        // string -> any
+        try {
+            return Literal::make(this->lexical_form(), target, node_storage);
+        } catch (std::runtime_error const &) {
+            return Literal{};
+        }
+    }
+
+    auto const *target_e = DatatypeRegistry::get_entry(target_dtid);
+    if (target_e == nullptr) {
+        // target not registered
+        return Literal{};
+    }
+
+    if (this_dtid == Boolean::datatype_id && target_e->numeric_ops.has_value()) {
+        // bool -> numeric
+        if (target_e->numeric_ops->is_impl()) {
+            auto const value = this->template value<Boolean>() ? target_e->numeric_ops->get_impl().one_value_fptr()
+                                                               : target_e->numeric_ops->get_impl().zero_value_fptr();
+
+            return Literal::make_typed_unchecked(target_e->to_string_fptr(value), target, node_storage);
+        } else {
+            auto const &impl_converter = DatatypeRegistry::get_numeric_op_impl_conversion(*target_e);
+            auto const *target_num_impl = DatatypeRegistry::get_numerical_ops(impl_converter.target_type_id);
+            assert(target_num_impl != nullptr);
+
+            // perform conversion as impl numeric type
+            auto const value = this->template value<Boolean>() ? target_num_impl->get_impl().one_value_fptr()
+                                                               : target_num_impl->get_impl().zero_value_fptr();
+
+            // downcast to target
+            auto const target_value = impl_converter.inverted_convert(value);
+
+            if (!target_value.has_value()) {
+                // not representable as target type
+                return Literal{};
+            }
+
+            return Literal::make_typed_unchecked(target_e->to_string_fptr(*target_value), target, node_storage);
+        }
+    }
+
+    auto const *this_e = DatatypeRegistry::get_entry(this_dtid);
+    if (this_e == nullptr) {
+        // this datatype not registered
+        return Literal{};
+    }
+
+    if (auto const common_conversion = DatatypeRegistry::get_common_type_conversion(this_e->conversion_table, target_e->conversion_table); common_conversion.has_value()) {
+        // general cast
+        // TODO: if performance is bad split into separate cases for up-, down- and cross-casting to avoid one set of std::any wrapping and unwrapping for the former 2
```

Comment:
I would cover this definitely in a separate PR. How much work is it? What's your estimate?

Reply:
almost no work, just looks uglier. Which is why I went with this version

Reply:
should I change it or leave it as is?

Reply:
leave it for now.